src/s3/util/fileViewerManager.ts

```diff
@@ -274,31 +306,65 @@ export class S3FileViewerManager {
-        showOutputMessage('-------contents in temp:', this.outputChannel)
-
-        fs.readdirSync(this.tempLocation!).forEach((file: any) => {
-            showOutputMessage(` ${file}`, this.outputChannel)
-            getLogger().debug(` ${file}`)
-        })
+    private async removeAndCloseTab(activeTab: S3Tab): Promise<void> {
+        let fileNode: S3FileNode | undefined = activeTab.s3FileNode
+        fileNode = await this.refreshNode(fileNode)
+        if (!fileNode) {
+            return
+        }
 
-        getLogger().debug('-------------------------')
-        showOutputMessage('-------------------------', this.outputChannel)
-        return Promise.resolve()
-    }*/
+        this.activeTabs.delete(activeTab.fileUri.fsPath)
+        this.cacheArns.delete(fileNode.arn)
+        await activeTab.focusAndCloseTab()
+        try {
+            fs.unlinkSync(await this.createTargetPath(fileNode))
+        } catch (e) {
+            showOutputMessage(e, this.outputChannel)
+        }
+    }
 
-    private async createTemp(): Promise<void> {
-        this.tempLocation = await makeTemporaryToolkitFolder()
+    public async createTemp(): Promise<string> {
+        this._tempLocation = await makeTemporaryToolkitFolder()
         showOutputMessage(
             localize(
                 'AWS.s3.message.tempCreation',
                 'Temp folder for FileViewer created with location: {0}',
-                this.tempLocation
+                this._tempLocation
             ),
             this.outputChannel
         )
-        getLogger().info(`Temp folder for FileViewer created with location: ${this.tempLocation}`)
+        getLogger().info(`Temp folder for FileViewer created with location: ${this._tempLocation}`)
+        return this._tempLocation
+    }
+
+    public get tempLocation(): string | undefined {
+        return this._tempLocation
+    }
+
+    /**
+     * Checks that the cached date is after the last modified date in S3.
+     * If not, file targetUri is invalid and needs to be redownloaded.
+     *
+     * @param fileNode instance in S3
+     * @param targetUri file downloaded in system
+     * @returns
+     */
+    private async checkForValidity(fileNode: S3FileNode, targetUri: vscode.Uri): Promise<boolean> {
+        const newNode = await this.refreshNode(fileNode)
+        if (!newNode) {
+            getLogger().error(`FileViewer: Error, refreshNode() returned undefined with file: ${fileNode.file.key}`)
+            getLogger().debug(`Cache contains: ${this.cacheArns.toString()}`)
+            return false
+        }
+
+        fileNode = newNode
+        const lastModifiedInS3 = fileNode!.file.lastModified
+        const { birthtime } = fs.statSync(targetUri.fsPath)
+
+        getLogger().debug(
+            `FileViewer: File ${fileNode.file.name} was last modified in S3: ${lastModifiedInS3}, cached on: ${birthtime}`
+        )
+
+        return lastModifiedInS3! <= birthtime
```

Comment:
It might be worth adding additional safety to this since this can throw an error if lastModifiedInS3 is undefined. Would be nice to have a getLogger().error() if this is undefined?